### PR TITLE
Init branch

### PR DESCRIPTION
**Pull request by MajdT51**

### CODE REVIEW

File: contracts/AddrArrayLib.sol [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+     */
+    function pushAddress(Addresses storage self, address element) internal {
+      if (!exists(self, element)) {
+        self._items.push(element);
+      }
+    }
+
+    /**
+     * @notice remove an address from the array
+     * @dev it wil find the element and then Swap it with the last element and then delete it
```
Comment (jpbu) [inline contracts/AddrArrayLib.sol]: it wil find the element and then Swap it with the last element and then delete it -> 
finds the element, swaps it with the last element, and then deletes it; returns a boolean whether the element was found and deleted

File: contracts/AddrArrayLib.sol [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        return false;
+    }
+
+    /**
+     * @notice get the address at a specific index from array
+     * @param self Storage array containing address type variables
+     * @param index the index in the array
+     */
+    function getAddressAtIndex(Addresses storage self, uint256 index) internal view returns (address) {
+        return self._items[index];
```
Comment (jpbu) [inline contracts/AddrArrayLib.sol]: What happens if the index is out of bounds (i.e. >= items.length)?

File: contracts/TokenContract.sol [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    uint256 private totalSupply;
+    mapping (address => uint256) balances;
+
+    constructor () public {
+        totalSupply = 100;
+        balances[msg.sender] = 100;
+    }
+
+    function mintToken(address to, uint256 amount) external {
+        require(trustedMinters.exists(msg.sender), 'The sender address is not registered as a Minter');
```
Comment (jpbu) [inline contracts/TokenContract.sol]: Minter -> minter

File: test/AddrArrayLib-test.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+const TokenContract = artifacts.require("TokenContract");
+
+contract('TokenContract', (accounts) => {
+
+  const owner = accounts[0];
+  let tokenContractInsatnce;
```
Comment (jpbu) [inline test/AddrArrayLib-test.js]: tokenContractInsatnce -> tokenContractInstance

File: test/AddrArrayLib-test.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+  it('should deploy the smart contrat and check total supply', async () => {
```
Comment (jpbu) [inline test/AddrArrayLib-test.js]: should deploy the smart contrat and check total supply -> should deploy the smart contract and set the correct initial total supply

File: test/AddrArrayLib-test.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    tokenContractInsatnce = await TokenContract.deployed();
+
+    let initTotalSupply = (await tokenContractInsatnce.getTotalSupply.call()).toNumber();
+    assert.equal(initTotalSupply, 100, "the init total supply was not 100");
+  });
+
+  it('should check that at the beginning the list of trusted minters is empty', async () => {
```
Comment (jpbu) [inline test/AddrArrayLib-test.js]: number of trusted minters should be 0 at the beginning

File: test/AddrArrayLib-test.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+    let numberOfTrustedMinters = (await tokenContractInsatnce.getNumberOfMinters.call()).toNumber();
+    assert.equal(numberOfTrustedMinters, 0, "the trusted minters array is not empty");
+  });
+
+  it('should add 3 addresses as minters to the first TokenContract', async () => {
```
Comment (jpbu) [inline test/AddrArrayLib-test.js]: why "first" TokenContract? Currently there is only one

File: test/AddrArrayLib-test.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+contract('TokenContract', (accounts) => {
+
+  const owner = accounts[0];
+  let tokenContractInsatnce;
+
+  it('should deploy the smart contrat and check total supply', async () => {
+    tokenContractInsatnce = await TokenContract.deployed();
+
+    let initTotalSupply = (await tokenContractInsatnce.getTotalSupply.call()).toNumber();
+    assert.equal(initTotalSupply, 100, "the init total supply was not 100");
```
Comment (jpbu) [inline test/AddrArrayLib-test.js]: Should either use single quotes or double quotes for strings, but not mixed

File: test/AddrArrayLib-test.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  });
+
+  it('should add 3 addresses as minters to the first TokenContract', async () => {
+
+    await tokenContractInsatnce.addMinterToTrustedList(accounts[1], {from: owner});
+    await tokenContractInsatnce.addMinterToTrustedList(accounts[2], {from: owner});
+    await tokenContractInsatnce.addMinterToTrustedList(accounts[3], {from: owner});
+
+    let numberOfTrustedMinters = (await tokenContractInsatnce.getNumberOfMinters.call()).toNumber();
+    assert.equal(numberOfTrustedMinters, 3, "the trusted minters array has more/less than 3 Minters");
```
Comment (jpbu) [inline test/AddrArrayLib-test.js]: Minters -> minters (several occurrences)

File: README.md [hunk 10]
```diff
@@ -1 +1,28 @@
...
+
+### Version
+    Truffle v5.0.10 (core: 5.0.10)
+    Solidity v0.5.0 (solc-js)
+    Node v8.11.3
+    Web3.js v1.0.0-beta.37
+    Ganache v2.1.0
+
+### Setup
+* Download Ganache https://github.com/trufflesuite/ganache/releases/v2.1.0
```
Comment (jpbu) [inline README.md]: Would be nice to provide an alternative with Ganache in Docker (a simple `docker run` command would suffice) for those (like me) who have trouble running the Ganache desktop version

File: contracts/AddrArrayLib.sol (hunk 2 above)
Comment (MajdT51) [reply contracts/AddrArrayLib.sol]: it will revert, I changed the revert msg to "the  index is out of bounds"

**Pull request merged**